### PR TITLE
[#2645] Use clojure.specs to retrieve visualisation spec column names

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**
- [x] test with empty (or minimum) [vizs](https://gist.github.com/tangrammer/416560648d8da36f40f349c29affa977)
- [x] test against prod tenant viz dbs

Relates #2645

### CODE REVIEW

**Comment by tangrammer** [comment]
Adapted and tested against all prod tenant vizs 
https://github.com/akvo/akvo-lumen/pull/2647/commits/a29003bac615620646aa55c61e7982ed57501968

File: backend/specs/akvo/lumen/specs/visualisation.clj [hunk 1]
```diff
@@ -68,63 +93,163 @@
...
+(alias 'pivot.s 'akvo.lumen.specs.visualisation.pivot)
+
+(s/def ::hideColumnTotals boolean?)
+(s/def ::hideRowTotals boolean?)
+(s/def ::rowColumn (s/nilable ::db.dsv.column.s/columnName))
+
+(defn str-nat-int? [v]
+  (when (some? v)
+    (try
+      (nat-int? (read-string v))
```
Comment (iperdomo) [inline backend/specs/akvo/lumen/specs/visualisation.clj:176]: Let's use a different strategy for converting to number, `read-string` should be avoided IMHO.

Comment (tangrammer) [reply backend/specs/akvo/lumen/specs/visualisation.clj]: changed https://github.com/akvo/akvo-lumen/pull/2647/commits/b8522d02ff6f7a3356d715413c54436f67886ee5

Comment (tangrammer) [reply backend/specs/akvo/lumen/specs/visualisation.clj]: and https://github.com/akvo/akvo-lumen/pull/2647/commits/84ea1be0d56de8c507501b43f149aab8bbe27edd

**Pull request merged**